nemo_skills/pipeline/utils.py

```diff
@@ -99,29 +101,279 @@ def get_registered_external_repo(name: str) -> Optional[RepoMetadata]:
     return EXTERNAL_REPOS[name]
 
 
+def is_mounted_filepath(cluster_config: dict, path: str):
+    """
+    Check if the filepath is mounted using the cluster config. Does not raise an error if the filepath is not mounted.
+
+    Args:
+        cluster_config: cluster config dictionary
+        path: path to the file to be mounted
+
+    Returns:
+        bool: Whether the filepath is mounted.
+    """
+    # Find which mount path matches the filepaths prefix
+    for mount in get_mounts_from_config(cluster_config) + ['/nemo_run/code:/nemo_run/code']:
+        mount_source, mount_dest = mount.split(':')
+        if path.startswith(mount_dest):
+            return True
+
+    return False
+
+
+
 def check_if_mounted(cluster_config, path_to_check):
     """Will check that path_to_check is referenced inside one of the mounts."""
     if cluster_config["executor"] == "none":
         # no mounts in local executor
         return
 
-    for mount in get_mounts_from_config(cluster_config) + ['/nemo_run/code:/nemo_run/code']:
-        if path_to_check.startswith(mount.split(":")[1]):
-            return
-    raise ValueError(f"The path '{path_to_check}' is not mounted. Check cluster config.")
+    if not is_mounted_filepath(cluster_config, path_to_check):
+        raise ValueError(f"The path '{path_to_check}' is not mounted. Check cluster config.")
+
+def get_mounted_path(cluster_config: dict, path: str):
+    """
+    Resolve the mounted filepath using the cluster config to merge the mount destination path to the filepath.
+
+    Args:
+        cluster_config:
```

Comment:
best to either remove this or add some description

Reply:
Done